layer/layer_store.go

```diff
@@ -563,7 +563,7 @@ func (ls *layerStore) ReleaseRWLayer(l RWLayer) ([]Metadata, error) {
 	}
 
 	if m.hasReferences() {
-		return []Metadata{}, nil
+		return []Metadata{}, ErrActiveMount
```

Comment:
I'm not sure that we really need to return an error here.
At this point the layer is released, just not cleaned up. I think that's ok.